base/linalg/triangular.jl

```diff
@@ -1095,77 +1244,19 @@ function logm{T<:Union{Float64,Complex{Float64}}}(A0::UpperTriangular{T})
         else
             logAk = log(Ak)
             logAkp1 = log(Akp1)
-            w = atanh((Akp1 - Ak)/(Akp1 + Ak)) + im*pi*ceil((imag(logAkp1-logAk)-pi)/(2*pi))
+            w = atanh((Akp1 - Ak)/(Akp1 + Ak)) + im * pi * unw(logAkp1-logAk)
             dd = 2 * exp(p*(logAk+logAkp1)/2) * sinh(p*w) / (Akp1 - Ak);
             A[k,k+1] = A0[k,k+1] * dd
         end
     end
+end
 
-    # Compute accurate diagonal of T
-    for i = 1:n
-        a = A0[i,i]
-        if s == 0
-            r = a - 1
-        end
-        s0 = s
-        if angle(a) >= pi / 2
-            a = sqrt(a)
-            s0 = s - 1
-        end
-        z0 = a - 1
-        a = sqrt(a)
-        r = 1 + a
-        for j = 1:s0-1
-            a = sqrt(a)
-            r = r * (1 + a)
-        end
-        A[i,i] = z0 / r
-    end
-
-    # Compute the Gauss-Legendre quadrature formula
-    R = zeros(Float64, m, m)
-    for i = 1:m - 1
-        R[i,i+1] = i / sqrt((2 * i)^2 - 1)
-        R[i+1,i] = R[i,i+1]
-    end
-    x,V = eig(R)
-    w = Array(Float64, m)
-    for i = 1:m
-        x[i] = (x[i] + 1) / 2
-        w[i] = V[1,i]^2
-    end
-
-    # Compute the Padé approximation
-    Y = zeros(T, n, n)
-    for k = 1:m
-        Y = Y + w[k] * (A / (x[k] * A + I))
-    end
-
-    # Scale back
-    scale!(2^s, Y)
-
-    # Compute accurate diagonal and superdiagonal of log(T)
-    for k = 1:n-1
-        Ak = A0[k,k]
-        Akp1 = A0[k+1,k+1]
-        logAk = log(Ak)
-        logAkp1 = log(Akp1)
-        Y[k,k] = logAk
-        Y[k+1,k+1] = logAkp1
-        if Ak == Akp1
-            Y[k,k+1] = A0[k,k+1] / Ak
-        elseif 2 * abs(Ak) < abs(Akp1) || 2 * abs(Akp1) < abs(Ak)
-            Y[k,k+1] = A0[k,k+1] * (logAkp1 - logAk) / (Akp1 - Ak)
-        else
-            w = atanh((Akp1 - Ak)/(Akp1 + Ak) + im*pi*(ceil((imag(logAkp1-logAk) - pi)/(2*pi))))
-            Y[k,k+1] = 2 * A0[k,k+1] * w / (Akp1 - Ak)
-        end
-    end
-
-    return UpperTriangular(Y)
-
+function unw{T<:Number}(x::T)
+    return (isreal(x)? zero(T) : T(ceil((imag(x) - pi) / (2 * pi))))
 end
-logm(A::LowerTriangular) = logm(A.').'
+@vectorize_1arg Number unw
```

Comment:
Why is there a vectorized version of this function?  We don't seem to use it, and the function is not exported.